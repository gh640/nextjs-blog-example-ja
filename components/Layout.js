import Head from "next/head"
import Link from "next/link"

const Layout = (props) => {
  const { title, children } = props
  const siteTitle = "後藤のブログ"

  return (
    <div className="page">
      <Head>
        <title>{title ? `${title} | ${siteTitle}` : siteTitle}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <header>
        <h1 className="site-title">
          <Link href="/">
            <a>{siteTitle}</a>
          </Link>
        </h1>
      </header>

      <main>
        {title ? <h1 className="page-title">{title}</h1> : ``}
        <div className="page-main">
          {children}
        </div>
      </main>

      <footer>
        &copy; {siteTitle}
      </footer>

      <style jsx>{`
        .page {
          padding: 2em 1em;
          max-width: 800px;
          margin-left: auto;
          margin-right: auto;
        }

        header {
          display: flex;
          justify-content: center;
          align-items: center;
          margin: 0 0 4em;
        }

        .site-title a {
          color: inherit;
          text-decoration: none;
        }

        footer {
          margin-top: 4em;
          padding-top: 2em;
          padding-bottom: 2em;
          display: flex;
          justify-content: center;
          align-items: center;
        }
      `}</style>

      <style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: 'Noto Sans JP', -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
          color: #222;
        }

        img,
        iframe {
          max-width: 100%;
        }

        h1, h2, h3, h4, h5, h6 {
          font-family: Montserrat, -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      `}</style>
    </div>
  )
}

export default Layout
