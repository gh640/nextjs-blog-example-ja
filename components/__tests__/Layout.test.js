import { cloneElement } from "react"
import Head from "next/head"
import Link from "next/link"
import renderer from "react-test-renderer"

import Layout from "../Layout.js"

jest.mock("next/head")
jest.mock("next/link")

describe("Layout", () => {
	Head.mockImplementation(({ children }) => children)
	Link.mockImplementation(({ href, children }) => 
		cloneElement(children, { href })
	)

	it("renders collectly with title", () => {
		const tree = renderer
			.create(<Layout title="ページタイトル"></Layout>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders collectly with children", () => {
		const tree = renderer
			.create(<Layout>
				<div>
					<span>Hello World</span>
				</div>
			</Layout>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
