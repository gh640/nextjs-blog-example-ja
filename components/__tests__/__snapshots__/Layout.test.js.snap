// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders collectly with children 1`] = `
<div
  className="page"
>
  <title>
    後藤のブログ
  </title>
  <link
    href="/favicon.ico"
    rel="icon"
  />
  <header>
    <h1
      className="site-title"
    >
      <a
        href="/"
      >
        後藤のブログ
      </a>
    </h1>
  </header>
  <main>
    
    <div
      className="page-main"
    >
      <div>
        <span>
          Hello World
        </span>
      </div>
    </div>
  </main>
  <footer>
    © 
    後藤のブログ
  </footer>
  <style
    jsx={true}
  >
    
        .page {
          padding: 2em 1em;
          max-width: 800px;
          margin-left: auto;
          margin-right: auto;
        }

        header {
          display: flex;
          justify-content: center;
          align-items: center;
          margin: 0 0 4em;
        }

        .site-title a {
          color: inherit;
          text-decoration: none;
        }

        footer {
          margin-top: 4em;
          padding-top: 2em;
          padding-bottom: 2em;
          display: flex;
          justify-content: center;
          align-items: center;
        }
      
  </style>
  <style
    global={true}
    jsx={true}
  >
    
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: 'Noto Sans JP', -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
          color: #222;
          font-size: 16px;
        }

        img,
        iframe {
          max-width: 100%;
        }

        h1, h2, h3, h4, h5, h6 {
          font-family: Montserrat, -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
        }

        /* Prism のコードブロックに影響を与えず &lt;code&gt; にスタイルを追加 */
        code:not([class]) {
          font-family: monospace;
          font-size: 1rem;
          background-color: #eee;
          display: inline-block;
          padding: .2em .4em;
        }

        * {
          box-sizing: border-box;
        }
      
  </style>
</div>
`;

exports[`Layout renders collectly with title 1`] = `
<div
  className="page"
>
  <title>
    ページタイトル | 後藤のブログ
  </title>
  <link
    href="/favicon.ico"
    rel="icon"
  />
  <header>
    <h1
      className="site-title"
    >
      <a
        href="/"
      >
        後藤のブログ
      </a>
    </h1>
  </header>
  <main>
    <h1
      className="page-title"
    >
      ページタイトル
    </h1>
    <div
      className="page-main"
    />
  </main>
  <footer>
    © 
    後藤のブログ
  </footer>
  <style
    jsx={true}
  >
    
        .page {
          padding: 2em 1em;
          max-width: 800px;
          margin-left: auto;
          margin-right: auto;
        }

        header {
          display: flex;
          justify-content: center;
          align-items: center;
          margin: 0 0 4em;
        }

        .site-title a {
          color: inherit;
          text-decoration: none;
        }

        footer {
          margin-top: 4em;
          padding-top: 2em;
          padding-bottom: 2em;
          display: flex;
          justify-content: center;
          align-items: center;
        }
      
  </style>
  <style
    global={true}
    jsx={true}
  >
    
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: 'Noto Sans JP', -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
          color: #222;
          font-size: 16px;
        }

        img,
        iframe {
          max-width: 100%;
        }

        h1, h2, h3, h4, h5, h6 {
          font-family: Montserrat, -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
        }

        /* Prism のコードブロックに影響を与えず &lt;code&gt; にスタイルを追加 */
        code:not([class]) {
          font-family: monospace;
          font-size: 1rem;
          background-color: #eee;
          display: inline-block;
          padding: .2em .4em;
        }

        * {
          box-sizing: border-box;
        }
      
  </style>
</div>
`;
